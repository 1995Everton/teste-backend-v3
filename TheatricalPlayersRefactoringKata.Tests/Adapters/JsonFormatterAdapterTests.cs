using ApprovalTests;
using ApprovalTests.Reporters;
using AutoMapper;
using System.Collections.Generic;
using TheatricalPlayersRefactoringKata.Application.Adapters;
using TheatricalPlayersRefactoringKata.Application.Mappings;
using TheatricalPlayersRefactoringKata.Entities;
using Xunit;

namespace TheatricalPlayersRefactoringKata.Tests.Adapters
{
    public class JsonFormatterAdapterTests
    {
        private readonly IMapper _mapper;

        public JsonFormatterAdapterTests()
        {
            var config = new MapperConfiguration(cfg => {
                cfg.AddProfile<MappingProfile>();
            });

            _mapper = config.CreateMapper();
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void Format_ShouldReturnFormattedStatement()
        {
            // Configura um cenário onde uma declaração é formatada em JSON.
            // Verifica se a declaração formatada está correta.

            // Arrange
            var formatter = new JsonFormatterAdapter(_mapper);
            var statement = new Statement
            {
                Customer = "John Doe",
                Items = new List<StatementItem>
                    {
                        new StatementItem { Name = "Hamlet", AmountOwed = 400.00m, Seats = 55 },
                        new StatementItem { Name = "As You Like It", AmountOwed = 300.00m, Seats = 35 },
                        new StatementItem { Name = "Othello", AmountOwed = 500.00m, Seats = 40 }
                    },
                TotalAmountOwed = 1200.00m,
                TotalEarnedCredits = 30
            };

            // Act
            var result = formatter.Format(statement);

            // Assert
            Approvals.Verify(result);
        }
    }
}

