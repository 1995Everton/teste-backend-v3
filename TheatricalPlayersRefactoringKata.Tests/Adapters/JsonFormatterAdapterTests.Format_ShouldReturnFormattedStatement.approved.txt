{
  "Customer": "John Doe",
  "Items": [
    {
      "AmountOwed": 400.00,
      "EarnedCredits": 0.0,
      "Seats": 55
    },
    {
      "AmountOwed": 300.00,
      "EarnedCredits": 0.0,
      "Seats": 35
    },
    {
      "AmountOwed": 500.00,
      "EarnedCredits": 0.0,
      "Seats": 40
    }
  ],
  "TotalAmountOwed": 1200.00,
  "TotalEarnedCredits": 30.0
}